src/main/java/io/xdb/core/process/ProcessStartConfig.java

```diff
@@ -0,0 +1,31 @@
+package io.xdb.core.process;
+
+import io.xdb.core.persistence.PersistenceTableRowToUpsert;
+import io.xdb.gen.models.ProcessIdReusePolicy;
+import java.util.HashMap;
+import java.util.Map;
+import lombok.Builder;
+import lombok.Getter;
+
+@Builder
+@Getter
+public class ProcessStartConfig {
+
+    private final int timeoutSeconds;
+    private final ProcessIdReusePolicy processIdReusePolicy;
+    /**
+     * table name: table row
+     */
+    private final Map<String, PersistenceTableRowToUpsert> globalAttributesToUpsert = new HashMap<>();
+
+    /**
+     * Add a table row as global attributes to upsert and return the updated config.
+     *
+     * @param tableRow  the table row.
+     * @return the updated config.
+     */
+    public ProcessStartConfig addGlobalAttributesToUpsert(final PersistenceTableRowToUpsert tableRow) {
```

Comment:
I actually tried to avoid calling it "upsert" because the behaviour could be insert/upsert based on different policy. 

https://github.com/xdblab/xdb-golang-samples/blob/5c657cd0a058c5bf7a91e12a65d02d989930bcdc/cmd/server/xdb/signup_controller.go#L41

This is for initial write. 

In state execute API and RPC, we will start with "update" only for now because Postgres doesn't have a way to fully implement upsert 🤦 https://github.com/xdblab/xdb/issues/90
That's why the xdb-apis does't say anything about "upsert"/"insert".